File: 250510/빙빙 돌며 사각형 채우기/snail-alphabet-square.java
import java.util.Scanner;
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();
        // Please write your code here.
        int charValue = 65;
        int count = 1;
        // 동남서북 0123
        int[] dx = {0,1,0,-1};
        int[] dy = {1,0,-1,0};
        int directionIdx = 0;
        char[][] map = new char[n][m];

        int x = 0;
        int y = -1;
        
        while (count <= n*m) {
            int nx = x + dx[directionIdx];
            int ny = y + dy[directionIdx];

            if (nx<0 || nx>=n || ny<0 || ny>=m || map[nx][ny]!=0) {
                directionIdx = (directionIdx + 1) % 4;
            }

            x = x + dx[directionIdx];
            y = y + dy[directionIdx];

            map[x][y] = (char) charValue;

            charValue++;
            count++;
            if (charValue > 90) {
                charValue = 65;
            }
        }

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                System.out.print(map[i][j] + " ");
            }
            System.out.println();
        }
    }
}

// 

File: 251021/빙빙 돌며 사각형 채우기/snail-alphabet-square.py
n, m = map(int, input().split())

# Please write your code here.

curAlphabet = 'A'

grid = [[0 for _ in range(m)] for _ in range(n)] 

curX = 0
curY = 0

# 동남서북
dx = [0,1,0,-1]
dy = [1,0,-1,0]

dir = 0

grid[curX][curY] = curAlphabet
# curAlphabet = 'B'

def inRange(x,y) :
    if x>=0 and x<n and y>=0 and y < m :
        return True
    return False

for i in range(1, n * m) :
    nx = curX + dx[dir]
    ny = curY + dy[dir]
    curAlphabet = chr(ord(curAlphabet) + 1)
    if curAlphabet > 'Z' :
        curAlphabet = 'A'

    while not inRange(nx,ny) or grid[nx][ny] != 0 :
        dir = (dir + 1) % 4
        nx = curX + dx[dir]
        ny = curY + dy[dir]

    grid[nx][ny] = curAlphabet

    curX = nx
    curY = ny

for i in range(n) :
    for j in range(m) :
        print(grid[i][j],end=" ")
    print()